### PR TITLE
patches: update schedule for May cycle and remove 1.18 schedule

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this:
/kind documentation

#### What this PR does / why we need it:
- Update patch releases schedule for May cycle
- remove schedule information for release 1.18 because we will not build any patch for this release anymore since reach the EOL

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

/assign @saschagrunert @hasheddan 
cc @kubernetes/release-engineering

### CODE REVIEW

File: releases/patch-releases.md [hunk 1]
```diff
@@ -82,6 +82,7 @@ End of Life for **1.19** is **2021-09-30**
 | PATCH RELEASE | CHERRY PICK DEADLINE | TARGET DATE |
 |--- |--- |--- |
 | 1.19.10       | 2021-04-09           | 2021-04-14  |
+| 1.19.10       | 2021-04-09           | 2021-04-14  |
```
Comment (hasheddan) [inline releases/patch-releases.md:85]: @cpanato i think you have a duplicate here

Comment (cpanato) [reply releases/patch-releases.md:85]: opss :) i'm blind

thanks for the catch

Comment (cpanato) [reply releases/patch-releases.md:85]: fixed

**Review by hasheddan — APPROVED** [review]
/lgtm

**Comment by puerco** [comment]
@cpanato yesterday during a conversation with @justaugustus we determined to have one last 1.18 patch release before EOL.

**Comment by cpanato** [comment]
@puerco ok, and that will be together with the other patches?

**Comment by puerco** [comment]
I think not, I think we'll cut it separately before the month is out. Let's discuss this at the next meeting.

**Pull request merged**